autoleagueplay/replays.py

```diff
@@ -2,24 +2,71 @@
 from enum import Enum
 from pathlib import Path
 from typing import Dict, Any
+import os
+import RattletrapPython.rattletrap as rat
+import json
 
 import requests
 from rlbottraining.history.metric import Metric
+from autoleagueplay.paths import WorkingDir
 from watchdog.events import LoggingEventHandler
 from watchdog.observers import Observer
 
-
 class ReplayPreference(Enum):
     SAVE = 'save'  # save to the default replays directory
     CALCULATED_GG = 'calculated_gg'  # save locally and upload to https://calculated.gg/
     IGNORE_REPLAY = 'ignore'
+    ANONYMIZE_REPLAY = 'anonym'
 
 
 def upload_to_calculated_gg(replay_path: Path):
     with open(replay_path, 'rb') as f:
         response = requests.post('https://calculated.gg/api/upload', files={'replays': f})
         print(f'upload response to {replay_path.name}: {response}')
 
+def anonymize_replay(replay_path: Path, working_dir: WorkingDir):
+    replay_name = str(os.path.basename(str(replay_path)).split('.')[0])
+    anonym_replay_path = working_dir.replays / str(replay_name + '.replay')
+    anonym_json_path = working_dir.replays / str('replay' + '.json')
+    rat.parse(str(replay_path), anonym_json_path)
+
+    with open(anonym_json_path) as replay_file:
+        replay = json.load(replay_file)
+    try:
+        replay['header']['body']['properties']['value']['Team0Score']['value']['int'] = 0
+    except:
+        pass
+    try:
+        replay['header']['body']['properties']['value']['Team1Score']['value']['int'] = 0
+    except:
+        pass
+    try:
+        replay['header']['body']['properties']['value']['HighLights']['value']['array'] = []
+    except:
+        pass
+    try:
+        replay['header']['body']['properties']['value']['HighLights']['size'] = '0'
+    except:
+        pass
+    try:
+        replay['header']['body']['properties']['value']['Goals']['value']['array'] = []
+    except:
+        pass
+    try:
+        replay['header']['body']['properties']['value']['Goals']['size'] = '0'
+    except:
+        pass
+    try:
+        replay['content']['body']['marks'] = []
+    except:
+        pass
```

Comment:
Just curious, does all these try-excepts has to be done individually?

Reply:
there may be a better way, but if one fails, i want it to keep going to the next ( for example, if team 0 doesn't score, that field doesn't exist)

Reply:
Then this is probably fine.